pystac/item.py

```diff
@@ -423,6 +423,9 @@ def from_dict(
         migrate: bool = False,
         preserve_dict: bool = True,
     ) -> T:
+        if preserve_dict:
```

Comment:
not a fan of the re-assign to a method param name, especially one just named `d`

Reply:
I just moved this block up in the method. As it was prior the dict would be altered if `migrate=True` which seemed to go against the intent of the kwarg.